### PR TITLE
feat(common): marker processing🙀

### PR DESCRIPTION
**Pull request by srl295**
- emit the sentinel value, with range checking

#9119

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by srl295** [comment]
the pt abnt2 keyboard is temporarily disabled.. it needs its transforms fleshed out, creates a marker validation err

File: common/web/types/src/ldml-keyboard/pattern-parser.ts [hunk 1]
```diff
@@ -40,6 +47,25 @@ export class MarkerParser {
...
 
+  /**
+   * Marker sentinel as a string - U+FFFF
+   */
+  public static readonly SENTINEL = String.fromCodePoint(constants.marker_sentinel);
+
+  /**
+   * Matches all markers.
+   */
+  public static readonly SENTINEL_ALL_MARKERS = this.SENTINEL + this.SENTINEL;
```
Comment (mcdurdin) [inline common/web/types/src/ldml-keyboard/pattern-parser.ts:58]: This makes me a little squirmy. I'm probably worrying about nothing but it does mean `SENTINEL` has two meanings, whereas in kmx processor `SENTINEL` only has a single meaning.

FWIW, in kmx processor, `SENTINEL` is always followed by a `CODE` to determine the actual meaning of the sentinel, e.g. `SENTINEL CODE_DEADKEY`. I wonder if we should be following the same pattern here to allow for future expansion

File: common/web/types/src/ldml-keyboard/pattern-parser.ts [hunk 2]
```diff
@@ -56,6 +82,35 @@ export class MarkerParser {
...
+  }
+
+  /** @returns all marker strings as sentinel values */
+  public static toSentinelString(s: string, markers?: OrderedStringList) : string {
+    if (!s) return s;
+    return s.replaceAll(this.REFERENCE, (sub, arg) => {
+      if (arg === MarkerParser.ANY_MARKER_ID) {
+        return MarkerParser.SENTINEL_ALL_MARKERS;
+      }
+      if (!markers) {
```
Comment (mcdurdin) [inline common/web/types/src/ldml-keyboard/pattern-parser.ts:101]: Given this, why is `markers` an optional parameter?

**Review by mcdurdin — APPROVED** [review]
LGTM

File: common/web/types/src/ldml-keyboard/pattern-parser.ts (hunk 2 above)
Comment (srl295) [reply common/web/types/src/ldml-keyboard/pattern-parser.ts:101]: May be no markers in that file.

File: common/web/types/src/ldml-keyboard/pattern-parser.ts (hunk 1 above)
Comment (srl295) [reply common/web/types/src/ldml-keyboard/pattern-parser.ts:58]: @mcdurdin i actually couldn't find `U_SENTINEL` or `SENTINEL` in the keyman source…  I did see SENTINEL_CODE_MARKER which is FDD0 but it didn't seem to go with any other sentinel value. 

I did delete SENTINEL_ALL_MARKERS in a future PR

Comment (mcdurdin) [reply common/web/types/src/ldml-keyboard/pattern-parser.ts:58]: Yep, we discussed this at standup but for reference, `UC_SENTINEL CODE_DEADKEY <marker>` is the sequence used by kmx.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.151-alpha](https://keyman.com/downloads/releases/alpha/17.0.151)

**Pull request merged**